tdvt/README.md

```diff
@@ -0,0 +1,42 @@
+# TDVT datasets
+
+## Import the datasets
+### Calcs
+```
+mongoimport --uri "{your connection URL}" --db=tdvt --drop --collection calcs --type=json --file={Path to tdvt/datasets directory}/Calcs.json
+```
+### Staples
+```
+mongoimport --uri "{your connection URL}" --db=tdvt --drop --collection calcs --type=json --file={Path to tdvt/datasets directory}/Staples.json
+```
+
+## Set the schemas
+
+### Make sure to be on the correct database
+```
+use tdvt
+```
+
+### Load Calcs schema
+```
+db.runCommand({ sqlSetSchema: "Calcs", schema: { "version": 1, "jsonSchema": { bsonType: ['object'], properties: { bool2: { bsonType: ['bool', 'null'] }, date3: { bsonType: ['date', 'null'] }, _id: { bsonType: ['objectId', 'null'] }, num2: { bsonType: ['double', 'null'] }, time1: { bsonType: ['date', 'null'] }, num4: { bsonType: ['double', 'null'] }, int1: { bsonType: ['null', 'int'] }, num1: { bsonType: ['double', 'null'] }, bool0: { bsonType: ['bool', 'null'] }, datetime0: { bsonType: ['date', 'null'] }, date2: { bsonType: ['date', 'null'] }, time0: { bsonType: ['date', 'null'] }, date0: { bsonType: ['null', 'date'] }, int3: { bsonType: ['int', 'null'] }, str0: { bsonType: ['string', 'null'] }, date1: { bsonType: ['date', 'null'] }, int0: { bsonType: ['int', 'null'] }, str3: { bsonType: ['string', 'null'] }, key: { bsonType: ['string', 'null'] }, num0: { bsonType: ['null', 'double'] }, datetime1: { bsonType: ['string', 'null'] }, str1: { bsonType: ['string', 'null'] }, int2: { bsonType: ['int', 'null'] }, zzz: { bsonType: ['string', 'null'] }, bool3: { bsonType: ['null', 'bool'] }, bool1: { bsonType: ['bool', 'null'] }, num3: { bsonType: ['double', 'null'] }, str2: { bsonType: ['string', 'null'] } } } } })
```

Comment:
Using the sqlGenerateSchema command does not create schemas matching exactly the tables definition found here:  https://github.com/tableau/connector-plugin-sdk/blob/master/tests/datasets/TestV1/postgres/load_calcs.sql and https://github.com/tableau/connector-plugin-sdk/blob/master/tests/datasets/TestV1/postgres/load_staples.sql

By default, if not specified otherwise, columns are Nullable in SQL that's why all Calcs columns are set nullable.

Reply:
Ah that's too bad they require it to be nullable by default even if the data doesn't contain nulls 😞  but that makes sense, thanks for explaining